### ISSUE TITLE
MessageBar - error messages and the alert role attribute

### ISSUE THREAD

**Issue by curtisj44** [microsoft/fluentui#16453]
### Environment Information

- **Package version(s)**: 7.155.3 (latest version as of today)
- **Browser and OS versions**: all

### The Issue

I'm trying to use a `MessageBar` to indicate errors are present when a form is submitted. If errors are present, the `MessageBar` will be displayed:

![image](https://user-images.githubusercontent.com/810231/104385718-98f0dd00-54e8-11eb-92ca-68efeea11d8c.png)

I was initially using this basic `MessageBar` markup:

```jsx
<MessageBar messageBarType={MessageBarType.error}>
  Invalid submission. Review the errors above.
</MessageBar>
```

Which outputs:

```html

<div class="ms-MessageBar ms-MessageBar--error ms-MessageBar-multiline">
  <div class="ms-MessageBar-content">
    {/* omitting `.ms-MessageBar-icon` for clarity */}

    <div class="ms-MessageBar-text" role="status" aria-live="assertive">
      <span class="ms-MessageBar-innerText">
        <span>Invalid submission. Review the errors above.</span>
      </span>
    </div>
  </div>
</div>
```

To bring the user's attention to the error immediately, I get the impression it's best to use [`role="alert"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Alert_Role). However, that doesn't seem possible. The `role="status"` is hardcoded in [MessageBar.base.tsx#L94](https://github.com/microsoft/fluentui/blob/master/packages/react-internal/src/components/MessageBar/MessageBar.base.tsx#L94).

Adding to my confusion, adding a `role` prop to `MessageBar` like this:

```jsx
<MessageBar
  messageBarType={MessageBarType.error}
  role="alert"
>
  Invalid submission. Review the errors above.
</MessageBar>
```

Appends the `role` to `.ms-MessageBar-innerText` (which is nested inside the current `role="status"`):

```html
<div class="ms-MessageBar ms-MessageBar--error ms-MessageBar-multiline">
  <div class="ms-MessageBar-content">
    {/* omitting `.ms-MessageBar-icon` for clarity */}

    <div class="ms-MessageBar-text" role="status" aria-live="assertive">
      <span class="ms-MessageBar-innerText" role="alert">
        <span>Content goes here.</span>
      </span>
    </div>
  </div>
</div>
```

### The Demo

I put together some different variation of `MessageBar` + `role` here: https://codepen.io/curtisj44/pen/RwGexOz

### The Questions

1. Am I off-base in hoping to use `role="alert"` with `MessageBar`?

1. Should the `MessageBar` support defining a different `role`?

1. Should an Error `MessageBar` automatically use `role="error"` instead of `role="status"`?

1. Is the currently output combination of `role="status"` and `aria-live="assertive"` on an Error `MessageBar` sufficient for optimal accessibility? (fwiw, I've heard it doesn't announce correctly on Chrome v87 + NVDA. Though, I now see that the library's "screen reader support is limited to Edge + Narrator")

**Comment by curtisj44** [comment]
cc: @smhigley

**Comment by gouttierre** [comment]
@curtisj44 - thanks for the feedback and the questions posted. I will ask our experienced engineer to assist you on your queries.

@khmakoto - could you kindly guide/ assist @curtisj44 on his open questions.
- [ ] Am I off-base in hoping to use role="alert" with MessageBar?
- [ ] Should the MessageBar support defining a different role?
- [ ] Should an Error MessageBar automatically use role="error" instead of role="status"?
- [ ] Is the currently output combination of role="status" and aria-live="assertive" on an Error MessageBar sufficient for optimal accessibility? (fwiw, I've heard it doesn't announce correctly on Chrome v87 + NVDA. Though, I now see that the library's "screen reader support is limited to Edge + Narrator")

**Comment by smhigley** [comment]
After a quick look, it looks like the Error, Blocked, and SevereWarning MessageBar types should all default to `role="alert"`. The rest make sense with `role="status"`.

If we make that change, I'm not sure if we'd also need to allow a role override in properties. I'd lean towards no for now, and see if anyone comes up with a use case where that breakdown of alert vs. status doesn't work.

**Comment by khmakoto** [comment]
That sounds good, thanks for that @smhigley!

@curtisj44 would you be willing to make a PR with this change?

**Comment by curtisj44** [comment]
@khmakoto Yes. I think I can take a look at implementing the changes @smhigley mentions 👍

**Comment by khmakoto** [comment]
@curtisj44 cool! Let us know once you have a PR we can take a look at!

**Comment by max-zheng** [comment]
I was following this thread because my team's website also uses `MessageBar` with type `error` and we are updating the site to make it WCAG 2.1 compliant. Just checking in to see if there are any new updates regarding this issue. Thanks.

**Comment by curtisj44** [comment]
I hope to have something up for review this week 🤞
